modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -831,18 +849,27 @@ private CompletableFuture<Void> saveRecoverableStateToMetastorage(
     ) {
         Operation[] puts = new Operation[3 + zoneIds.size()];
 
-        puts[0] = put(zonesNodesAttributes(), toBytes(nodesAttributes()));
+        puts[0] = put(zonesNodesAttributes(), VersionedSerialization.toBytes(nodesAttributes(), NodesAttributesSerializer.INSTANCE));
 
         puts[1] = put(zonesRecoverableStateRevision(), longToBytesKeepingOrder(revision));
 
-        puts[2] = put(zonesLastHandledTopology(), toBytes(newLogicalTopology));
+        puts[2] = put(
+                zonesLastHandledTopology(),
+                VersionedSerialization.toBytes(newLogicalTopology, LogicalTopologySetSerializer.INSTANCE)
+        );
 
         int i = 3;
 
         // TODO: https://issues.apache.org/jira/browse/IGNITE-19491 Properly utilise topology augmentation map. Also this map
         // TODO: can be saved only once for all zones.
         for (Integer zoneId : zoneIds) {
-            puts[i++] = put(zoneTopologyAugmentation(zoneId), toBytes(zonesState.get(zoneId).topologyAugmentationMap()));
+            puts[i++] = put(
+                    zoneTopologyAugmentation(zoneId),
+                    VersionedSerialization.toBytes(
```

Comment:
Constructions like 
```
                    VersionedSerialization.toBytes(
                            zonesState.get(zoneId).topologyAugmentationMap(),
                            TopologyAugmentationMapSerializer.INSTANCE
                    )
```
look a bit overloaded. 

Maybe, we could hide it inside `TopologyAugmentationMapSerializer`, so it could look like 
```
TopologyAugmentationMapSerializer.toBytes(zonesState.get(zoneId).topologyAugmentationMap())
``` 
?
WDYT? Could this idea be adapted to all Serializers/Deserializers?

Reply:
Applied this approach where it makes sense